Index: AWSCodeOSS/build-private/src/src/tsconfig.base.json
===================================================================
--- AWSCodeOSS.orig/build-private/src/src/tsconfig.base.json
+++ AWSCodeOSS/build-private/src/src/tsconfig.base.json
@@ -5,7 +5,7 @@
 		"experimentalDecorators": true,
 		"noImplicitReturns": true,
 		"noImplicitOverride": true,
-		"noUnusedLocals": true,
+		"noUnusedLocals": false,
 		"allowUnreachableCode": false,
 		"strict": true,
 		"exactOptionalPropertyTypes": false,
Index: AWSCodeOSS/build-private/src/src/vs/editor/test/node/classification/typescript.test.ts
===================================================================
--- AWSCodeOSS.orig/build-private/src/src/vs/editor/test/node/classification/typescript.test.ts
+++ AWSCodeOSS/build-private/src/src/vs/editor/test/node/classification/typescript.test.ts
@@ -108,7 +108,7 @@ function parseTest(fileName: string): IT
 	return { content, assertions };
 }
 
-// @ts-expect-error
+// @ts-ignore
 function executeTest(fileName: string, parseFunc: IParseFunc): void {
 	const { content, assertions } = parseTest(fileName);
 	const actual = parseFunc(content);
Index: AWSCodeOSS/build-private/src/src/vs/workbench/contrib/preferences/browser/preferencesSearch.ts
===================================================================
--- AWSCodeOSS.orig/build-private/src/src/vs/workbench/contrib/preferences/browser/preferencesSearch.ts
+++ AWSCodeOSS/build-private/src/src/vs/workbench/contrib/preferences/browser/preferencesSearch.ts
@@ -30,7 +30,7 @@ export interface IEndpointDetails {
 export class PreferencesSearchService extends Disposable implements IPreferencesSearchService {
 	declare readonly _serviceBrand: undefined;
 
-	// @ts-expect-error disable remote search for now, ref https://github.com/microsoft/vscode/issues/172411
+	// @ts-ignore disable remote search for now, ref https://github.com/microsoft/vscode/issues/172411
 	private _installedExtensions: Promise<ILocalExtension[]>;
 	private _remoteSearchProvider: IRemoteSearchProvider | undefined;
 
