Index: AWSCodeOSS/build-private/src/src/vs/server/node/webClientServer.ts
===================================================================
--- AWSCodeOSS.orig/build-private/src/src/vs/server/node/webClientServer.ts
+++ AWSCodeOSS/build-private/src/src/vs/server/node/webClientServer.ts
@@ -166,7 +166,7 @@ export class WebClientServer {
 			return serveError(req, res, 400, `Bad request.`);
 		}
 
-		return serveFile(filePath, this._environmentService.isBuilt ? CacheControl.NO_EXPIRY : CacheControl.ETAG, this._logService, req, res, headers);
+		return serveFile(filePath, CacheControl.ETAG, this._logService, req, res, headers);
 	}
 
 	private _getResourceURLTemplateAuthority(uri: URI): string | undefined {
