Index: AWSCodeOSS/build/private/code-editor-src/src/tsconfig.base.json
===================================================================
--- AWSCodeOSS.orig/build/private/code-editor-src/src/tsconfig.base.json
+++ AWSCodeOSS/build/private/code-editor-src/src/tsconfig.base.json
@@ -6,7 +6,7 @@
 		"experimentalDecorators": true,
 		"noImplicitReturns": true,
 		"noImplicitOverride": true,
-		"noUnusedLocals": true,
+		"noUnusedLocals": false,
 		"noUncheckedSideEffectImports": true,
 		"allowUnreachableCode": false,
 		"strict": true,
Index: AWSCodeOSS/build/private/code-editor-src/src/vs/workbench/contrib/preferences/browser/preferencesSearch.ts
===================================================================
--- AWSCodeOSS.orig/build/private/code-editor-src/src/vs/workbench/contrib/preferences/browser/preferencesSearch.ts
+++ AWSCodeOSS/build/private/code-editor-src/src/vs/workbench/contrib/preferences/browser/preferencesSearch.ts
@@ -30,7 +30,7 @@ export interface IEndpointDetails {
 export class PreferencesSearchService extends Disposable implements IPreferencesSearchService {
 	declare readonly _serviceBrand: undefined;
 
-	// @ts-expect-error disable remote search for now, ref https://github.com/microsoft/vscode/issues/172411
+	// @ts-ignore disable remote search for now, ref https://github.com/microsoft/vscode/issues/172411
 	private _installedExtensions: Promise<ILocalExtension[]>;
 	private _remoteSearchProvider: IRemoteSearchProvider | undefined;
 	private _aiSearchProvider: IAiSearchProvider | undefined;
Index: AWSCodeOSS/build/private/code-editor-src/src/vs/workbench/contrib/remote/browser/tunnelView.ts
===================================================================
--- AWSCodeOSS.orig/build/private/code-editor-src/src/vs/workbench/contrib/remote/browser/tunnelView.ts
+++ AWSCodeOSS/build/private/code-editor-src/src/vs/workbench/contrib/remote/browser/tunnelView.ts
@@ -762,7 +762,7 @@ export class TunnelPanel extends ViewPan
 	private protocolChangableContextKey: IContextKey<boolean>;
 	private isEditing: boolean = false;
 	// TODO: Should this be removed?
-	//@ts-expect-error
+	//@ts-ignore
 	private titleActions: IAction[] = [];
 	private lastFocus: number[] = [];
 
