Index: third-party-src/product.json
===================================================================
--- third-party-src.orig/product.json
+++ third-party-src/product.json
@@ -1,18 +1,19 @@
 {
-	"nameShort": "Code - OSS",
-	"nameLong": "Code - OSS",
-	"applicationName": "code-oss",
-	"dataFolderName": ".vscode-oss",
+	"nameShort": "Code Editor",
+	"nameLong": "Code Editor",
+	"codeEditorVersion": "1.0.0",
+	"applicationName": "code",
+	"dataFolderName": ".vscode-editor",
 	"win32MutexName": "vscodeoss",
 	"licenseName": "MIT",
-	"licenseUrl": "https://github.com/microsoft/vscode/blob/main/LICENSE.txt",
-	"serverLicenseUrl": "https://github.com/microsoft/vscode/blob/main/LICENSE.txt",
+	"licenseUrl": "https://github.com/aws/code-editor/blob/main/LICENSE",
+	"serverLicenseUrl": "https://github.com/aws/code-editor/blob/main/LICENSE",
 	"serverGreeting": [],
 	"serverLicense": [],
 	"serverLicensePrompt": "",
-	"serverApplicationName": "code-server-oss",
-	"serverDataFolderName": ".vscode-server-oss",
-	"tunnelApplicationName": "code-tunnel-oss",
+	"serverApplicationName": "code-editor-server",
+	"serverDataFolderName": ".code-editor-server",
+	"tunnelApplicationName": "code-editor-tunnel",
 	"win32DirName": "Microsoft Code OSS",
 	"win32NameVersion": "Microsoft Code OSS",
 	"win32RegValueName": "CodeOSS",
@@ -27,9 +28,9 @@
 	"darwinBundleIdentifier": "com.visualstudio.code.oss",
 	"darwinProfileUUID": "47827DD9-4734-49A0-AF80-7E19B11495CC",
 	"darwinProfilePayloadUUID": "CF808BE7-53F3-46C6-A7E2-7EDB98A5E959",
-	"linuxIconName": "code-oss",
+	"linuxIconName": "code-editor",
 	"licenseFileName": "LICENSE.txt",
-	"reportIssueUrl": "https://github.com/microsoft/vscode/issues/new",
+	"reportIssueUrl": "https://github.com/aws/code-editor/issues/new",
 	"nodejsRepository": "https://nodejs.org",
 	"urlProtocol": "code-oss",
 	"webviewContentExternalBaseUrlTemplate": "https://{{uuid}}.vscode-cdn.net/insider/ef65ac1ba57f57f2a3961bfe94aa20481caca4c6/out/vs/workbench/contrib/webview/browser/pre/",
Index: third-party-src/src/vs/base/common/product.ts
===================================================================
--- third-party-src.orig/src/vs/base/common/product.ts
+++ third-party-src/src/vs/base/common/product.ts
@@ -60,6 +60,7 @@ export interface IProductConfiguration {
 	readonly date?: string;
 	readonly quality?: string;
 	readonly commit?: string;
+	readonly codeEditorVersion?: string;
 
 	readonly nameShort: string;
 	readonly nameLong: string;
Index: third-party-src/src/vs/workbench/browser/parts/dialogs/dialogHandler.ts
===================================================================
--- third-party-src.orig/src/vs/workbench/browser/parts/dialogs/dialogHandler.ts
+++ third-party-src/src/vs/workbench/browser/parts/dialogs/dialogHandler.ts
@@ -79,9 +79,9 @@ export class BrowserDialogHandler extend
 	async about(): Promise<void> {
 		const detailString = (useAgo: boolean): string => {
 			return localize('aboutDetail',
-				"Version: {0}\nCommit: {1}\nDate: {2}\nBrowser: {3}",
+				"Code Editor Version: {0}\nCode - OSS Version: {1}\nDate: {2}\nBrowser: {3}",
+				this.productService.codeEditorVersion || 'Unknown',
 				this.productService.version || 'Unknown',
-				this.productService.commit || 'Unknown',
 				this.productService.date ? `${this.productService.date}${useAgo ? ' (' + fromNow(new Date(this.productService.date), true) + ')' : ''}` : 'Unknown',
 				navigator.userAgent
 			);
Index: third-party-src/src/vs/platform/product/common/product.ts
===================================================================
--- third-party-src.orig/src/vs/platform/product/common/product.ts
+++ third-party-src/src/vs/platform/product/common/product.ts
@@ -61,8 +61,8 @@ else {
 	if (Object.keys(product).length === 0) {
 		Object.assign(product, {
 			version: '1.95.0-dev',
-			nameShort: 'Code - OSS Dev',
-			nameLong: 'Code - OSS Dev',
+			nameShort: 'Code Editor',
+			nameLong: 'Code Editor',
 			applicationName: 'code-oss',
 			dataFolderName: '.vscode-oss',
 			urlProtocol: 'code-oss',
