Index: third-party-src/src/vs/code/electron-sandbox/workbench/workbench-dev.html
===================================================================
--- third-party-src.orig/src/vs/code/electron-sandbox/workbench/workbench-dev.html
+++ third-party-src/src/vs/code/electron-sandbox/workbench/workbench-dev.html
@@ -26,7 +26,7 @@
 				;
 				script-src
 					'self'
-					'unsafe-eval'
+					'wasm-unsafe-eval'
 					blob:
 					'nonce-0c6a828f1297'
 				;
Index: third-party-src/src/vs/code/electron-sandbox/workbench/workbench.html
===================================================================
--- third-party-src.orig/src/vs/code/electron-sandbox/workbench/workbench.html
+++ third-party-src/src/vs/code/electron-sandbox/workbench/workbench.html
@@ -26,7 +26,7 @@
 				;
 				script-src
 					'self'
-					'unsafe-eval'
+					'wasm-unsafe-eval'
 					blob:
 				;
 				style-src
Index: third-party-src/src/vs/server/node/webClientServer.ts
===================================================================
--- third-party-src.orig/src/vs/server/node/webClientServer.ts
+++ third-party-src/src/vs/server/node/webClientServer.ts
@@ -423,7 +423,7 @@ export class WebClientServer {
 			'default-src \'self\';',
 			'img-src \'self\' https: data: blob:;',
 			'media-src \'self\';',
-			`script-src 'self' 'unsafe-eval' ${WORKBENCH_NLS_BASE_URL ?? ''} blob: 'nonce-1nline-m4p' ${this._getScriptCspHashes(data).join(' ')} '${webWorkerExtensionHostIframeScriptSHA}' 'sha256-/r7rqQ+yrxt57sxLuQ6AMYcy/lUpvAIzHjIJt/OeLWU=' ${useTestResolver ? '' : `http://${remoteAuthority}`};`,  // the sha is the same as in src/vs/workbench/services/extensions/worker/webWorkerExtensionHostIframe.html
+			`script-src 'self' 'wasm-unsafe-eval' ${WORKBENCH_NLS_BASE_URL ?? ''} blob: 'nonce-1nline-m4p' ${this._getScriptCspHashes(data).join(' ')} '${webWorkerExtensionHostIframeScriptSHA}' 'sha256-/r7rqQ+yrxt57sxLuQ6AMYcy/lUpvAIzHjIJt/OeLWU=' ${useTestResolver ? '' : `http://${remoteAuthority}`};`,  // the sha is the same as in src/vs/workbench/services/extensions/worker/webWorkerExtensionHostIframe.html
 			'child-src \'self\';',
 			`frame-src 'self' https://*.vscode-cdn.net data:;`,
 			'worker-src \'self\' data: blob:;',
Index: third-party-src/src/vs/workbench/contrib/notebook/browser/view/renderers/backLayerWebView.ts
===================================================================
--- third-party-src.orig/src/vs/workbench/contrib/notebook/browser/view/renderers/backLayerWebView.ts
+++ third-party-src/src/vs/workbench/contrib/notebook/browser/view/renderers/backLayerWebView.ts
@@ -319,7 +319,7 @@ export class BackLayerWebView<T extends ICommonCellInfo> extends Themable {
 				${enableCsp ?
 				`<meta http-equiv="Content-Security-Policy" content="
 					default-src 'none';
-					script-src ${webviewGenericCspSource} 'unsafe-inline' 'unsafe-eval';
+					script-src ${webviewGenericCspSource} 'unsafe-inline' 'wasm-unsafe-eval';
 					style-src ${webviewGenericCspSource} 'unsafe-inline';
 					img-src ${webviewGenericCspSource} https: http: data:;
 					font-src ${webviewGenericCspSource} https:;
Index: third-party-src/src/vs/workbench/services/extensions/worker/webWorkerExtensionHostIframe.html
===================================================================
--- third-party-src.orig/src/vs/workbench/services/extensions/worker/webWorkerExtensionHostIframe.html
+++ third-party-src/src/vs/workbench/services/extensions/worker/webWorkerExtensionHostIframe.html
@@ -4,7 +4,7 @@
 		<meta http-equiv="Content-Security-Policy" content="
 			default-src 'none';
 			child-src 'self' data: blob:;
-			script-src 'self' 'unsafe-eval' 'sha256-cl8ijlOzEe+0GRCQNJQu2k6nUQ0fAYNYIuuKEm72JDs=' https: http://localhost:* blob:;
+			script-src 'self' 'wasm-unsafe-eval' 'sha256-cl8ijlOzEe+0GRCQNJQu2k6nUQ0fAYNYIuuKEm72JDs=' https: http://localhost:* blob:;
 			connect-src 'self' https: wss: http://localhost:* http://127.0.0.1:* ws://localhost:* ws://127.0.0.1:*;"/>
 	</head>
 	<body>
