Index: third-party-src/src/vs/base/common/network.ts
===================================================================
--- third-party-src.orig/src/vs/base/common/network.ts
+++ third-party-src/src/vs/base/common/network.ts
@@ -220,7 +220,9 @@ class RemoteAuthoritiesImpl {
 		return URI.from({
 			scheme: platform.isWeb ? this._preferredWebSchema : Schemas.vscodeRemoteResource,
 			authority: `${host}:${port}`,
-			path: this._remoteResourcesPath,
+			path: platform.isWeb
+				? (window.location.pathname + "/" + this._remoteResourcesPath).replace(/\/\/+/g, "/")
+				: this._remoteResourcesPath,
 			query
 		});
 	}
Index: third-party-src/src/vs/base/common/product.ts
===================================================================
--- third-party-src.orig/src/vs/base/common/product.ts
+++ third-party-src/src/vs/base/common/product.ts
@@ -56,6 +56,7 @@ export type ExtensionVirtualWorkspaceSup
 };
 
 export interface IProductConfiguration {
+	readonly rootEndpoint?: string;
 	readonly version: string;
 	readonly date?: string;
 	readonly quality?: string;
Index: third-party-src/src/vs/code/browser/workbench/workbench-dev.html
===================================================================
--- third-party-src.orig/src/vs/code/browser/workbench/workbench-dev.html
+++ third-party-src/src/vs/code/browser/workbench/workbench-dev.html
@@ -63,7 +63,7 @@
 	</script>
 	<!-- Startup (do not modify order of script tags!) -->
 	<script>
-		const baseUrl = new URL('{{WORKBENCH_WEB_BASE_URL}}', window.location.origin).toString();
+		const baseUrl = new URL('{{WORKBENCH_WEB_BASE_URL}}', window.location).toString();
 		globalThis._VSCODE_FILE_ROOT = baseUrl + '/out/';
 	</script>
 	<script>
Index: third-party-src/src/vs/code/browser/workbench/workbench.html
===================================================================
--- third-party-src.orig/src/vs/code/browser/workbench/workbench.html
+++ third-party-src/src/vs/code/browser/workbench/workbench.html
@@ -60,7 +60,7 @@
 	</script>
 	<!-- Startup (do not modify order of script tags!) -->
 	<script>
-		const baseUrl = new URL('{{WORKBENCH_WEB_BASE_URL}}', window.location.origin).toString();
+		const baseUrl = new URL('{{WORKBENCH_WEB_BASE_URL}}', window.location).toString();
 		globalThis._VSCODE_FILE_ROOT = baseUrl + '/out/';
 	</script>
 	<script>
Index: third-party-src/src/vs/code/browser/workbench/workbench.ts
===================================================================
--- third-party-src.orig/src/vs/code/browser/workbench/workbench.ts
+++ third-party-src/src/vs/code/browser/workbench/workbench.ts
@@ -332,7 +332,8 @@ class LocalStorageURLCallbackProvider ex
 			this.startListening();
 		}
 
-		return URI.parse(mainWindow.location.href).with({ path: this._callbackRoute, query: queryParams.join('&') });
+		const path = (mainWindow.location.pathname + "/" + this._callbackRoute).replace(/\/\/+/g, "/");
+		return URI.parse(mainWindow.location.href).with({ path: path, query: queryParams.join('&') });
 	}
 
 	private startListening(): void {
Index: third-party-src/src/vs/platform/extensionResourceLoader/common/extensionResourceLoader.ts
===================================================================
--- third-party-src.orig/src/vs/platform/extensionResourceLoader/common/extensionResourceLoader.ts
+++ third-party-src/src/vs/platform/extensionResourceLoader/common/extensionResourceLoader.ts
@@ -120,7 +120,7 @@ export abstract class AbstractExtensionR
 					: version,
 				path: 'extension'
 			}));
-			return this._isWebExtensionResourceEndPoint(uri) ? uri.with({ scheme: RemoteAuthorities.getPreferredWebSchema() }) : uri;
+			return this._isWebExtensionResourceEndPoint(uri) ? URI.joinPath(URI.parse(window.location.href), uri.path) : uri;
 		}
 		return undefined;
 	}
Index: third-party-src/src/vs/server/node/webClientServer.ts
===================================================================
--- third-party-src.orig/src/vs/server/node/webClientServer.ts
+++ third-party-src/src/vs/server/node/webClientServer.ts
@@ -245,7 +245,10 @@ export class WebClientServer {
 		};
 
 		// Prefix routes with basePath for clients
-		const basePath = getFirstHeader('x-forwarded-prefix') || this._basePath;
+		const proxyPath = this._environmentService.args["base-path"] || "/";
+		const base = relativeRoot(proxyPath);
+		const vscodeBase = relativePath(proxyPath);
+		const basePath = vscodeBase || getFirstHeader("x-forwarded-prefix") || this._basePath;
 
 		const queryConnectionToken = parsedUrl.query[connectionTokenQueryName];
 		if (typeof queryConnectionToken === 'string') {
@@ -287,7 +290,7 @@ export class WebClientServer {
 		let remoteAuthority = (
 			useTestResolver
 				? 'test+test'
-				: (getFirstHeader('x-original-host') || getFirstHeader('x-forwarded-host') || req.headers.host)
+				: (getFirstHeader('x-original-host') || getFirstHeader('x-forwarded-host') || req.headers.host || window.location.host)
 		);
 		if (!remoteAuthority) {
 			return serveError(req, res, 400, `Bad request.`);
@@ -333,6 +336,7 @@ export class WebClientServer {
 		} : undefined;
 
 		const productConfiguration: Partial<Mutable<IProductConfiguration>> = {
+			rootEndpoint: base,
 			embedderIdentifier: 'server-distro',
 			extensionsGallery: this._webExtensionResourceUrlTemplate && this._productService.extensionsGallery ? {
 				...this._productService.extensionsGallery,
@@ -360,7 +364,7 @@ export class WebClientServer {
 		const workbenchWebConfiguration = {
 			remoteAuthority,
 			userDataPath: this._environmentService.userDataPath,
-			serverBasePath: basePath,
+			serverBasePath: this._basePath,
 			_wrapWebWorkerExtHostInIframe,
 			developmentOptions: { enableSmokeTestDriver: this._environmentService.args['enable-smoke-test-driver'] ? true : undefined, logLevel: this._logService.getLevel() },
 			settingsSyncOptions: !this._environmentService.isBuilt && this._environmentService.args['enable-sync'] ? { enabled: true } : undefined,
@@ -387,7 +391,9 @@ export class WebClientServer {
 			WORKBENCH_AUTH_SESSION: authSessionInfo ? asJSON(authSessionInfo) : '',
 			WORKBENCH_WEB_BASE_URL: staticRoute,
 			WORKBENCH_NLS_URL,
-			WORKBENCH_NLS_FALLBACK_URL: `${staticRoute}/out/nls.messages.js`
+			WORKBENCH_NLS_FALLBACK_URL: `${staticRoute}/out/nls.messages.js`,
+			BASE: base,
+			VS_BASE: vscodeBase
 		};
 
 		// DEV ---------------------------------------------------------------------------------------
@@ -497,3 +503,60 @@ export class WebClientServer {
 		return void res.end(data);
 	}
 }
+
+
+/**
+ * Remove extra slashes in a URL.
+ *
+ * This is meant to fill the job of `path.join` so you can concatenate paths and
+ * then normalize out any extra slashes.
+ *
+ * If you are using `path.join` you do not need this but note that `path` is for
+ * file system paths, not URLs.
+ */
+export const normalizeUrlPath = (url: string, keepTrailing = false): string => {
+	return url.replace(/\/\/+/g, "/").replace(/\/+$/, keepTrailing ? "/" : "")
+}
+
+/**
+ * Get the relative path that will get us to the root of the page. For each
+ * slash we need to go up a directory.  Will not have a trailing slash.
+ *
+ * For example:
+ *
+ * / => .
+ * /foo => .
+ * /foo/ => ./..
+ * /foo/bar => ./..
+ * /foo/bar/ => ./../..
+ *
+ * All paths must be relative in order to work behind a reverse proxy since we
+ * we do not know the base path.  Anything that needs to be absolute (for
+ * example cookies) must get the base path from the frontend.
+ *
+ * All relative paths must be prefixed with the relative root to ensure they
+ * work no matter the depth at which they happen to appear.
+ *
+ * For Express `req.originalUrl` should be used as they remove the base from the
+ * standard `url` property making it impossible to get the true depth.
+ */
+export const relativeRoot = (originalUrl: string): string => {
+	const depth = (originalUrl.split("?", 1)[0].match(/\//g) || []).length
+	return normalizeUrlPath("./" + (depth > 1 ? "../".repeat(depth - 1) : ""))
+}
+
+/**
+ * Get the relative path to the current resource.
+ *
+ * For example:
+ *
+ * / => .
+ * /foo => ./foo
+ * /foo/ => .
+ * /foo/bar => ./bar
+ * /foo/bar/ => .
+ */
+export const relativePath = (originalUrl: string): string => {
+	const parts = originalUrl.split("?", 1)[0].split("/")
+	return normalizeUrlPath("./" + parts[parts.length - 1])
+}
Index: third-party-src/src/vs/server/node/serverEnvironmentService.ts
===================================================================
--- third-party-src.orig/src/vs/server/node/serverEnvironmentService.ts
+++ third-party-src/src/vs/server/node/serverEnvironmentService.ts
@@ -91,6 +91,9 @@ export const serverOptions: OptionDescri
 
 	'compatibility': { type: 'string' },
 
+	/* ----- code-editor ----- */
+	'base-path': { type: 'string' },
+
 	_: OPTIONS['_']
 };
 
@@ -217,6 +220,9 @@ export interface ServerParsedArgs {
 
 	compatibility: string;
 
+	/* ----- code-editor ----- */
+	'base-path'?: string,
+
 	_: string[];
 }
 
Index: third-party-src/src/vs/platform/remote/browser/browserSocketFactory.ts
===================================================================
--- third-party-src.orig/src/vs/platform/remote/browser/browserSocketFactory.ts
+++ third-party-src/src/vs/platform/remote/browser/browserSocketFactory.ts
@@ -281,6 +281,7 @@ export class BrowserSocketFactory implem
 	connect({ host, port }: WebSocketRemoteConnection, path: string, query: string, debugLabel: string): Promise<ISocket> {
 		return new Promise<ISocket>((resolve, reject) => {
 			const webSocketSchema = (/^https:/.test(mainWindow.location.href) ? 'wss' : 'ws');
+			path = (mainWindow.location.pathname + "/" + path).replace(/\/\/+/g, "/")
 			const socket = this._webSocketFactory.create(`${webSocketSchema}://${(/:/.test(host) && !/\[/.test(host)) ? `[${host}]` : host}:${port}${path}?${query}&skipWebSocketFrames=false`, debugLabel);
 			const errorListener = socket.onError(reject);
 			socket.onOpen(() => {
