Index: code-editor-src/src/vs/server/node/webClientServer.ts
===================================================================
--- code-editor-src.orig/src/vs/server/node/webClientServer.ts
+++ code-editor-src/src/vs/server/node/webClientServer.ts
@@ -10,6 +10,7 @@ import * as path from 'path';
 import * as url from 'url';
 import * as cookie from 'cookie';
 import * as crypto from 'crypto';
+import { spawn } from 'child_process';
 import { isEqualOrParent } from '../../base/common/extpath.js';
 import { getMediaMime } from '../../base/common/mime.js';
 import { isLinux } from '../../base/common/platform.js';
@@ -40,6 +41,10 @@ const textMimeType: { [ext: string]: str
 	'.svg': 'image/svg+xml',
 };
 
+const enum ServiceName {
+	SAGEMAKER_UNIFIED_STUDIO = 'SageMakerUnifiedStudio',
+}
+
 /**
  * Return an error to the client.
  */
@@ -100,6 +105,7 @@ const STATIC_PATH = `/static`;
 const CALLBACK_PATH = `/callback`;
 const WEB_EXTENSION_PATH = `/web-extension-resource`;
 const IDLE_EXTENSION_PATH = `/api/idle`;
+const POST_STARTUP_SCRIPT_PATH = `/api/poststartup`;
 
 /**
  * Checks for terminal activity by reading the /dev/pts directory and comparing modification times of the files.
@@ -172,6 +178,9 @@ export class WebClientServer {
 			if (pathname === IDLE_EXTENSION_PATH) {
 				return this._handleIdle(req, res);
 			}
+			if (pathname === POST_STARTUP_SCRIPT_PATH) {
+				return this._handlePostStartupScriptInvocation(req, res);
+			}
 			if (pathname.startsWith(WEB_EXTENSION_PATH) && pathname.charCodeAt(WEB_EXTENSION_PATH.length) === CharCode.Slash) {
 				// extension resource support
 				return this._handleWebExtensionResource(req, res, pathname.substring(WEB_EXTENSION_PATH.length));
@@ -571,6 +580,41 @@ export class WebClientServer {
 			serveError(req, res, 500, error.message)
 		}
 	}
+
+	/**
+	 * Handles API requests to run the post-startup script in SMD.
+	 */
+	private async _handlePostStartupScriptInvocation(req: http.IncomingMessage, res: http.ServerResponse): Promise<void> {
+		const postStartupScriptPath = '/etc/sagemaker-ui/sagemaker_ui_post_startup.sh'
+		const logPath = '/var/log/apps/post_startup_default.log';
+		const logStream = fs.createWriteStream(logPath, { flags: 'a' });
+
+		// Only trigger post-startup script invocation for SageMakerUnifiedStudio app.
+		if (process.env['SERVICE_NAME'] != ServiceName.SAGEMAKER_UNIFIED_STUDIO) {
+			return serveError(req, res, 403, 'Forbidden');
+		} else {
+			//If postStartupScriptFile doesn't exist, it will throw FileNotFoundError (404)
+			//If exists, it will start the execution and add the execution logs in logFile.
+			try {
+				if (fs.existsSync(postStartupScriptPath)) {
+					// Adding 0o755 to make script file executable
+					fs.chmodSync(postStartupScriptPath, 0o755);
+
+					const subprocess = spawn('bash', [`${postStartupScriptPath}`], { cwd: '/' });
+					subprocess.stdout.pipe(logStream);
+					subprocess.stderr.pipe(logStream);
+
+					res.statusCode = 200;
+					res.setHeader('Content-Type', 'application/json');
+					res.end(JSON.stringify({ 'success': 'true' }));
+				} else {
+					serveError(req, res, 500, 'Poststartup script file not found at ' + postStartupScriptPath);
+				}
+			} catch (error) {
+				serveError(req, res, 500, error.message);
+			}
+		}
+	}
 }
 
 
