Index: third-party-src/build/gulpfile.vscode.web.js
===================================================================
--- third-party-src.orig/build/gulpfile.vscode.web.js
+++ third-party-src/build/gulpfile.vscode.web.js
@@ -188,6 +188,10 @@ function packageTask(sourceFolderName, d
 			gulp.src('resources/server/code-192.png', { base: 'resources/server' }),
 			gulp.src('resources/server/code-512.png', { base: 'resources/server' })
 		);
+		const indexHtml = gulp.src('resources/server/index.html', { base: 'resources/server' });
+
+		const productJsonStream = gulp.src(['product.json'], { base: '.' })
+			.pipe(json({ commit, date: readISODate('out-build'), version }));
 
 		const all = es.merge(
 			packageJsonStream,
@@ -196,7 +200,9 @@ function packageTask(sourceFolderName, d
 			deps,
 			favicon,
 			manifest,
-			pwaicons
+			pwaicons,
+			indexHtml,
+			productJsonStream
 		);
 
 		const result = all
