Index: third-party-src/src/vs/workbench/contrib/chat/browser/chat.contribution.ts
===================================================================
--- third-party-src.orig/src/vs/workbench/contrib/chat/browser/chat.contribution.ts
+++ third-party-src/src/vs/workbench/contrib/chat/browser/chat.contribution.ts
@@ -714,22 +714,6 @@ registerWorkbenchContribution2(ChatTrans
 registerWorkbenchContribution2(ChatContextContributions.ID, ChatContextContributions, WorkbenchPhase.AfterRestored);
 registerWorkbenchContribution2(ChatResponseResourceFileSystemProvider.ID, ChatResponseResourceFileSystemProvider, WorkbenchPhase.AfterRestored);
 
-registerChatActions();
-registerChatCopyActions();
-registerChatCodeBlockActions();
-registerChatCodeCompareBlockActions();
-registerChatFileTreeActions();
-registerChatTitleActions();
-registerChatExecuteActions();
-registerQuickChatActions();
-registerChatExportActions();
-registerMoveActions();
-registerNewChatActions();
-registerChatContextActions();
-registerChatDeveloperActions();
-registerChatEditorActions();
-registerChatToolActions();
-
 registerEditorFeature(ChatPasteProvidersFeature);
 
 
